neqo-transport/src/path.rs

```diff
@@ -560,7 +554,8 @@ impl Path {
         qlog: NeqoQlog,
         now: Instant,
     ) -> Self {
-        let mut sender = PacketSender::new(cc, pacing, Self::mtu_by_addr(remote.ip()), now);
+        let pmtud = PmtudState::new(remote.ip());
+        let mut sender = PacketSender::new(cc, pacing, pmtud.clone(), now);
```

Comment:
I couldn't come up with a better way to make the PMTU state available to both the path and the sender. Any better ideas?

Reply:
In general `Rc<RefCell<Foo>>` (interior mutability) is used across the codebase. Thus I assume this is a good start.

**Subjective**: I have shied away from interior mutability when writing Rust in the past. I find code that can be mutated by different entities (e.g. each with a reference to the same `RefCell`) hard to reason about. In addition, one can no longer depend on the borrow checker at compile time, but instead only finds out about issues at runtime. Obviously there are benefits to it though, e.g. instead of passing updates across entities, state is simply updated. Thus far, I find neqo to strike a reasonable balance.

In case you want to dig deeper, I can recommend the Rust book:

https://doc.rust-lang.org/book/ch15-05-interior-mutability.html#refcellt-and-the-interior-mutability-pattern

Especially the section combining `Rc` and `RefCell`:

https://doc.rust-lang.org/book/ch15-05-interior-mutability.html#having-multiple-owners-of-mutable-data-by-combining-rct-and-refcellt

In this particular case, given that `PacketSender` is owned by `Path`, why not have `Path` access `PmtudState` through its `PacketSender`? In other words, why not have `Path` call `self.sender.some_pmtud_state_function`?

Hope I didn't misunderstand the question.

Reply:
> In this particular case, given that PacketSender is owned by Path, why not have Path access PmtudState through its PacketSender? In other words, why not have Path call self.sender.some_pmtud_state_function?

I thought about this more, and I don't see how I could make that work, because also `ClassicCongestionControl` and `Pacer` (which are both owned by `PacketSender`) need access to the `PmtudState`. Any ideas?